# Crimes in Chicago 01/01/2023 - 05/11/2023

## Questions
- Find the total number of crimes recorded in the CRIME table.
- List community areas with per capita income less than 11000.
- List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis)
- List all kidnapping crimes involving a child?
- What kind of crimes were recorded at schools?
- List the average safety score for all types of schools.
- List 5 community areas with highest % of households below poverty line.
- Which community area(number) is most crime prone?
- Use a sub-query to find the name of the community area with highest hardship index.
- Use a sub-query to determine the Community Area Name with most number of crimes?

## Data
### Crimes Data
- id: ID of the person.
- iucr: four-digit codes that law enforcement agencies use to classify criminal incidents when taking individual reports.
- primary_type: the type of crime committed by a person.
- community_area: the ID of the area where the person resides.
- case_number: the case number assigned to the person's case. the Chicago Police Department RD Number (Record Division Number), which is unique to the incident.
- description: a detailed description of the crime.
- location_description: a description of the location where the crime occurred.

### Socio-Economic Data
- community_area_number: ID of the area.
- community_area_name: name of the community.
- per_capita_income: income per capita.
- percent_households_below_poverty: percentage of households below the poverty line.
- hardship_index: index of hardship.

### Source:
https://data.cityofchicago.org/browse?category=Public%20Safety

In [1]:
import pandas as pd
import sqlite3

# Transfering DF to SQLite

## Loading the SQL and creating the database.

In [2]:
%load_ext sql

In [3]:
%sql sqlite:///chicago_crimes.db

## Crimes Dataframe

In [4]:
df_crimes = pd.read_csv(r'./csv/chicago_crimes_2023.csv').drop('Unnamed: 0', axis = 1)
df_crimes.sample(3)

id case_number                    date  iucr primary_type  \
78688  13061447    JG246841  04/30/2023 05:20:00 PM  0810        THEFT   
29606  12983895    JG154134  02/15/2023 07:30:00 PM  0810        THEFT   
49733  13014222    JG190452  03/18/2023 05:20:00 PM  0530      ASSAULT   

                               description  \
78688                            OVER $500   
29606                            OVER $500   
49733  AGGRAVATED - OTHER DANGEROUS WEAPON   

                             location_description  community_area  
78688                                   APARTMENT              24  
29606  AIRPORT TERMINAL LOWER LEVEL - SECURE AREA              56  
49733                          GROCERY FOOD STORE              57

### Crimes Table 

- id: ID of the person.
- iucr: four-digit codes that law enforcement agencies use to classify criminal incidents when taking individual reports.
- primary_type: the type of crime committed by a person.
- community_area: the ID of the area where the person resides.
- case_number: the case number assigned to the person's case.
- description: a detailed description of the crime.
- location_description: a description of the location where the crime occurred.

In [5]:
%%sql sqlite://

CREATE TABLE crimes ( 
    id INTEGER PRIMARY KEY,
    case_number TEXT, 
    date DATE,
    iucr INTEGER, 
    primary_type TEXT,
    description TEXT,
    location_description TEXT,
    community_area INTEGER
)

Done.


[]

In [6]:
%%sql sqlite://

PRAGMA TABLE_INFO(crimes)

Done.


[(0, 'id', 'INTEGER', 0, None, 1),
 (1, 'case_number', 'TEXT', 0, None, 0),
 (2, 'date', 'DATE', 0, None, 0),
 (3, 'iucr', 'INTEGER', 0, None, 0),
 (4, 'primary_type', 'TEXT', 0, None, 0),
 (5, 'description', 'TEXT', 0, None, 0),
 (6, 'location_description', 'TEXT', 0, None, 0),
 (7, 'community_area', 'INTEGER', 0, None, 0)]

In [7]:
conn = sqlite3.connect('chicago_crimes.db')
df_crimes.to_sql('crimes', conn, if_exists='append', index=False)

86607

In [8]:
%%sql sqlite://

SELECT * 
FROM crimes
ORDER BY RANDOM() 
LIMIT 3

Done.


[(12993568, 'JG164776', '02/24/2023 08:23:00 PM', 1780, 'OFFENSE INVOLVING CHILDREN', 'OTHER OFFENSE', 'APARTMENT', 15),
 (12980212, 'JG149363', '02/11/2023 10:35:00 PM', '143A', 'WEAPONS VIOLATION', 'UNLAWFUL POSSESSION - HANDGUN', 'SIDEWALK', 68),
 (13033296, 'JG212891', '04/05/2023 09:00:00 PM', 1242, 'DECEPTIVE PRACTICE', 'COMPUTER FRAUD', 'APARTMENT', 6)]

## Socio-Economics Data Frame

In [9]:
df_economics = pd.read_csv(r'./csv/chicago_economics.csv').drop('Unnamed: 0', axis = 1)
df_economics.sample(5)

community_area_number community_area_name  \
24                   25.0              Austin   
45                   46.0       South Chicago   
25                   26.0  West Garfield Park   
59                   60.0          Bridgeport   
76                   77.0           Edgewater   

    percent_households_below_poverty  per_capita_income  hardship_index  
24                              28.6              15957            73.0  
45                              29.8              16579            75.0  
25                              41.7              10934            92.0  
59                              18.9              22694            43.0  
76                              18.2              33385            19.0

### Socio-Economics Table
- community_area_number: ID of the area.
- community_area_name: name of the community.
- per_capita_income: income per capita.
- percent_households_below_poverty: percentage of households below the poverty line.
- hardship_index: index of hardship.

In [10]:
%%sql sqlite://

CREATE TABLE economics ( 
    community_area_number INTEGER PRIMARY KEY,
    community_area_name TEXT, 
    percent_households_below_poverty REAL ,
    per_capita_income INTEGER , 
    hardship_index REAL
)

Done.


[]

In [11]:
%%sql sqlite://

PRAGMA TABLE_INFO(economics)

Done.


[(0, 'community_area_number', 'INTEGER', 0, None, 1),
 (1, 'community_area_name', 'TEXT', 0, None, 0),
 (2, 'percent_households_below_poverty', 'REAL', 0, None, 0),
 (3, 'per_capita_income', 'INTEGER', 0, None, 0),
 (4, 'hardship_index', 'REAL', 0, None, 0)]

In [12]:
conn = sqlite3.connect('chicago_crimes.db')
df_economics.to_sql('economics', conn, if_exists='append', index=False)

77

In [13]:
%%sql sqlite://

SELECT * 
FROM economics
ORDER BY RANDOM() 
LIMIT 5

Done.


[(55, 'Hegewisch', 17.1, 22677, 44.0),
 (40, 'Washington Park', 42.1, 13785, 88.0),
 (42, 'Woodlawn', 30.7, 18672, 58.0),
 (49, 'Roseland', 19.8, 17949, 52.0),
 (72, 'Beverly', 5.1, 39523, 12.0)]

# Exploratory Data Analysis

## Questions to answer:
- Find the total number of crimes recorded in the CRIME table.
- List community areas with per capita income less than 11000.
- List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis)
- List all kidnapping crimes involving a child?
- What kind of crimes were recorded at schools?
- List the average safety score for all types of schools.
- List 5 community areas with highest % of households below poverty line.
- Which community area(number) is most crime prone?
- Use a sub-query to find the name of the community area with highest hardship index.
- Use a sub-query to determine the Community Area Name with most number of crimes?

**Find the total number of crimes recorded in the CRIME table**

_Answer: there are 86,607 records from 1/01/2023 to 5/11/2023_

In [14]:
%%sql sqlite://

SELECT COUNT(*) AS crime_records
FROM crimes

Done.


[(86607,)]

**List community areas with per capita income less than 11000.**

_Answer: West Garfield Park, South Lawndale, Fuller Park, Riverdale	community has less than 11,000 income_

In [15]:
%%sql sqlite://

SELECT community_area_number, community_area_name, per_capita_income
FROM economics
WHERE per_capita_income < 11000

Done.


[(26, 'West Garfield Park', 10934),
 (30, 'South Lawndale', 10402),
 (37, 'Fuller Park', 10432),
 (54, 'Riverdale', 8201)]

**List all case numbers for crimes involving minors?(children are not considered minors for the purposes of crime analysis)**

In [16]:
%%sql sqlite://

SELECT id, case_number, primary_type, description
FROM crimes
WHERE description LIKE "%minor" COLLATE NOCASE

Done.


[(12951095, 'JG114028', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (12951330, 'JG115117', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (12958759, 'JG123650', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (12959000, 'JG118997', 'OBSCENITY', 'SALE / DISTRIBUTE OBSCENE MATERIAL TO MINOR'),
 (12962590, 'JG128232', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (12964635, 'JG130750', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (12970164, 'JG137500', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (12972039, 'JG139778', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (12977976, 'JG146801', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (12985982, 'JG156460', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (12995197, 'JG167355', 'LIQUOR LAW VIOLATION', 'ILLEGAL CONSUMPTION BY MINOR'),
 (12998127, 'JG171086', 'OBSCENITY', 'SALE / DISTRIBUTE OBSCENE MATERIAL TO MINOR'),
 (12999287, 'JG171896', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13006460, 'JG181067', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13007087, 'JG181817', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13012261, 'JG188141', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13013303, 'JG189345', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13013999, 'JG189529', 'LIQUOR LAW VIOLATION', 'ILLEGAL CONSUMPTION BY MINOR'),
 (13019525, 'JG196683', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13019538, 'JG196624', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13021672, 'JG199255', 'LIQUOR LAW VIOLATION', 'ILLEGAL POSSESSION BY MINOR'),
 (13022310, 'JG199683', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13024290, 'JG202339', 'OBSCENITY', 'SALE / DISTRIBUTE OBSCENE MATERIAL TO MINOR'),
 (13024427, 'JG202551', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13025627, 'JG203912', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13032790, 'JG212336', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13033828, 'JG213553', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13046585, 'JG228737', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13050272, 'JG233237', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13053569, 'JG237179', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13063327, 'JG248823', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13063811, 'JG249503', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13067620, 'JG254256', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13068808, 'JG255693', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR'),
 (13070078, 'JG257201', 'LIQUOR LAW VIOLATION', 'SELL / GIVE / DELIVER LIQUOR TO MINOR')]

**List all kidnapping crimes involving a child?**

In [17]:
%%sql sqlite://

SELECT id, case_number, primary_type, description
FROM crimes
WHERE primary_type LIKE "%kidnapping%" and description LIKE "%child%" COLLATE NOCASE

Done.


[(12955625, 'JG119981', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER'),
 (12956371, 'JG120818', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER'),
 (12966379, 'JG132742', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER'),
 (12997495, 'JG167961', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER'),
 (13025213, 'JG203175', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER'),
 (13052755, 'JG236194', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER'),
 (13059596, 'JG244548', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER'),
 (13061841, 'JG246935', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER'),
 (13066764, 'JG253121', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER'),
 (13067671, 'JG254360', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER'),
 (13068636, 'JG254898', 'KIDNAPPING', 'CHILD ABDUCTION / STRANGER')]

**What kind of crimes were recorded at schools?**

_Answer: Within the realm of schools, a comprehensive analysis of recorded crimes reveals intriguing insights. Topping the list is the crime of BATTERY, with a staggering **830 cases** reported. Following closely behind is the offense of ASSAULT, with **371 cases**. Not far behind, THEFT is recorded with **173 cases**. Collectively, these three types of crimes constitute a significant portion, accounting for a remarkable **65.30%** of the total crimes reported within school premises._

In [18]:
%%sql sqlite://
    
SELECT 
    primary_type,
    COUNT(primary_type) AS cases,
    ROUND(COUNT(primary_type) * 100.0 / (SELECT COUNT(*) FROM crimes WHERE location_description LIKE "%school%"), 2) AS rate
FROM crimes
WHERE location_description LIKE "%school%"
GROUP BY primary_type
ORDER BY cases DESC

Done.


[('BATTERY', 830, 45.13),
 ('ASSAULT', 371, 20.17),
 ('THEFT', 173, 9.41),
 ('OTHER OFFENSE', 109, 5.93),
 ('CRIMINAL DAMAGE', 75, 4.08),
 ('OFFENSE INVOLVING CHILDREN', 48, 2.61),
 ('SEX OFFENSE', 39, 2.12),
 ('CRIMINAL TRESPASS', 37, 2.01),
 ('NARCOTICS', 30, 1.63),
 ('PUBLIC PEACE VIOLATION', 23, 1.25),
 ('MOTOR VEHICLE THEFT', 23, 1.25),
 ('DECEPTIVE PRACTICE', 18, 0.98),
 ('CRIMINAL SEXUAL ASSAULT', 18, 0.98),
 ('BURGLARY', 17, 0.92),
 ('WEAPONS VIOLATION', 16, 0.87),
 ('ROBBERY', 7, 0.38),
 ('INTIMIDATION', 2, 0.11),
 ('HOMICIDE', 1, 0.05),
 ('CONCEALED CARRY LICENSE VIOLATION', 1, 0.05),
 ('ARSON', 1, 0.05)]

**List 5 community areas with highest % of households below poverty line.**

_Answer: Riverdale, Fuller Park, Englewood, North Lawndale, East Garfield Park has the highest % of households below poverty line._

In [23]:
%%sql sqlite://
    
SELECT
    community_area_number,
    community_area_name,
    percent_households_below_poverty
FROM economics
ORDER BY percent_households_below_poverty DESC
LIMIT 5

Done.


[(54, 'Riverdale', 56.5),
 (37, 'Fuller Park', 51.2),
 (68, 'Englewood', 46.6),
 (29, 'North Lawndale', 43.1),
 (27, 'East Garfield Park', 42.4)]

**Which community area(number) is most crime prone?**

_Answer: **Austin** (25) is the most crime prone having a total of **4254** crime cases._

In [20]:
%%sql sqlite://
    
SELECT 
    c.community_area,
    e.community_area_name,
    COUNT(*) AS crime_cases
FROM crimes AS c
LEFT JOIN economics AS e ON c.community_area = e.community_area_number
GROUP BY community_area
ORDER BY crime_cases DESC
LIMIT 5

Done.


[(25, 'Austin', 4254),
 (8, 'Near North Side', 3694),
 (28, 'Near West Side', 3456),
 (43, 'South Shore', 3170),
 (32, 'Loop', 2886)]

**Use a sub-query to find the name of the community area with highest hardship index.**

_Answer: Riverdale has the highest hardship index at 98_

In [21]:
%%sql sqlite://
    
SELECT community_area_name, hardship_index
FROM economics
WHERE hardship_index = (
    SELECT MAX(hardship_index)
    FROM economics
)

Done.


[('Riverdale', 98.0)]

**Use a sub-query to determine the Community Area Name with most number of crimes?**

_Answer: Austin has the most number of crimes at 4254 cases_

In [22]:
%%sql sqlite://
    
SELECT e.community_area_name, c.community_area, COUNT(c.primary_type) as crime_cases
FROM crimes AS c
LEFT JOIN economics AS e ON c.community_area = e.community_area_number
GROUP BY c.community_area
HAVING COUNT(c.primary_type) = (
    SELECT MAX(crime_count)
    FROM (
        SELECT COUNT(primary_type) AS crime_count
        FROM crimes
        GROUP BY community_area
    )
)

Done.


[('Austin', 25, 4254)]